lib/ui/window/platform_configuration.cc

```diff
@@ -396,4 +427,22 @@ std::string PlatformConfigurationNativeApi::DefaultRouteName() {
       ->DefaultRouteName();
 }
 
+int64_t PlatformConfigurationNativeApi::GetRootIsolateId() {
+  UIDartState* dart_state = UIDartState::Current();
+  FML_DCHECK(dart_state);
+  return dart_state->GetRootIsolateId();
+}
+
+void PlatformConfigurationNativeApi::RegisterBackgroundIsolate(
+    int64_t isolate_id) {
+  UIDartState* dart_state = UIDartState::Current();
+  FML_DCHECK(dart_state && !dart_state->IsRootIsolate());
+  auto weak_platform_message_handler =
+      (*static_cast<std::shared_ptr<PlatformMessageHandlerStorage>*>(
+           Dart_CurrentIsolateGroupData()))
```

Comment:
I added a check here.  I think it's very unlikely to get hit.  The only time it would be null is:
1) The function was called artificially from c++ (they'd have to make sure there was a DartState too)
1) The calling isolate isn't in the same group as the root isolate.  The public API makes this a bit tricky to do since you would have to send the root isolate token across isolate groups.